keepvariable/keepvariable_core.py

```diff
@@ -309,18 +319,20 @@ def release(self):
 
         return DummyLock()
 
-    def set(self, key, value, additional_params: Optional[dict] = None):
-        if additional_params is None:
-            additional_params = {}
+    def set(self, name: str, value, additional_params: Optional[dict] = None) -> dict[str, str]:
+        additional_params = {} if additional_params is None else additional_params
 
         value = self.parse_saved_value(value, additional_params)
-        self.storage[key] = value
-        return {key: value}
-
-    def get(self, key: str) -> Union[dict, pd.DataFrame, np.ndarray, datetime.datetime]:
-        value = self.storage.get(key)
-        decoded_value = self.decode_loaded_value(value)
-        return decoded_value
+        self.storage[name] = value
+        return {name: value}
+
+    # QUESTION: For in-app memory, is there a point to store serialized values?
+    def get(self, name: str) -> Union[dict, pd.DataFrame, np.ndarray, datetime.datetime]:
+        value = self.storage.get(name)
+        if isinstance(value, str):
+            decoded_value = self.decode_loaded_value(value)
+            return decoded_value
+        return value
```

Comment:
in which cases do we store np.ndarray?

Reply:
Forloop doesnt, but that doesnt mean we shouldnt be able to